pkg/controller/deployment/deployment_controller.go

```diff
@@ -749,3 +913,15 @@ func (dc *DeploymentController) updateDeployment(deployment *extensions.Deployme
 	// TODO: Using client for now, update to use store when it is ready.
 	return dc.expClient.Deployments(deployment.ObjectMeta.Namespace).Update(deployment)
 }
+
+func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deployment, rc *api.ReplicationController) (*extensions.Deployment, error) {
+	dTemplateHash := podutil.GetPodTemplateSpecHash(deployment.Spec.Template)
```

Comment:
Instead of hashing twice and comparing the hashes, just use reflect.DeepEqual. I did some benchmarking in exactly these two comparisons in the perma-failed PR and reflection is x10 times faster. See https://github.com/kubernetes/kubernetes/pull/19343#issuecomment-171996558